shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -396,17 +396,24 @@ - (void)setupChannels {
 
 - (void)maybeSetupPlatformViewChannels {
   if (_shell && self.shell.IsSetup()) {
+    fml::WeakPtr<FlutterEngine> weakSelf = [self getWeakPtr];
```

Comment:
This translation unit does not use ARC. So there should be no implicit capture self. Not sure this is doing anything.

Reply:
> This translation unit does not use ARC. So there should be no implicit capture self. Not sure this is doing anything.

`_platformPlugin` retain `self` implicitly. My test case would failed if we don't use `weakSelf`. 🤔

Reply:
@chinmaygarde this doesn't have to do with ARC, it has to do with the semantics of captured variables when creating blocks.  Since these blocks are using instance variables they are implicitly retaining self.